zdm/MCMC.py

```diff
@@ -0,0 +1,141 @@
+"""
+File: MCMC.py
+Author: Jordan Hoffmann
+Date: 28/09/23
+Purpose: 
+    Contains functions used for MCMC runs of the zdm code. MCMC_wrap.py is the 
+    main function which does the calling and this holds functions which do the 
+    MCMC analysis.
+"""
+
+import numpy as np
+
+import zdm.iteration as it
+
+import emcee
+import scipy.stats as st
+import pickle
+import time
+
+import multiprocessing as mp
+
+from zdm.misc_functions import *
+
+#==============================================================================
+
+def mcmc_likelihoods(outfile:str, walkers:int, steps:int, params, surveys, grids, ncpus=1):
+    posterior_sample = mcmc_runner(calc_log_posterior, outfile, params, surveys, grids, nwalkers=walkers, nsteps=steps, ncpus=ncpus)
```

Comment:
Jordan: can you please add some doc string for these functions? E.g. format of:
"""
[documentation]
"""
Specifically, give an overview of the function, then describe what the inputs are for? E.g. is "walkers" the integer number of walkers? Is there a suggested number (O~1, 10, 100...)? etc.

See some other functions for descriptions. Not everything here is fully documented, but I think this file is a pretty important one to specify as fully as possible.